Core/include/Acts/EventData/MultiTrajectory.hpp

```diff
@@ -928,25 +928,25 @@ class MultiTrajectory {
   template <typename T, HashedString key>
   constexpr T& component(IndexType istate) {
     assert(has(key, istate));
-    return *static_cast<T*>(self().template component_impl<key>(istate));
+    return *std::any_cast<T*>(self().component_impl(key, istate));
```

Comment:
Could you do:
```
auto comp = std::any_cast<T*>(self().component_impl(key, istate));
assert(comp);
return *comp;
```
so as not to duplicate the type check.

I didn't yet look at the implementation of `has`, so maybe it does more than that and you need both.

Reply:
`has` checks the `switch` and optionally the dynamic columns. You think this could be used to communicate if a component exists "in-band" (also cf. @tboldagh's comment)?